File: example-blocking/src/main/java/com/davin111/exampleblocking/client/MembershipClient.java
package com.davin111.exampleblocking.client;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.client.WebClient;

@Service
@RequiredArgsConstructor
public class MembershipClient {

    private WebClient.Builder webClientBuilder;

//    private WebClient webClient = webClientBuilder.build();

    public MembershipResponse getMembershipResponse(Long userId) {
        try {
            Thread.sleep(5000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        return new MembershipResponse(true);
    }
}

File: example-nonblocking-kotlin/src/main/kotlin/com/davin111/examplenonblockingkotlin/client/MembershipClient.kt
package com.davin111.examplenonblockingkotlin.client

import kotlinx.coroutines.delay
import org.springframework.stereotype.Service
import org.springframework.web.reactive.function.client.WebClient

@Service
class MembershipClient(
    private val webClientBuilder: WebClient.Builder,
) {

    suspend fun getMembershipResponse(userId: Long?): MembershipResponse {
        delay(5000)
        return MembershipResponse(true)
    }
}
